src/main/scala/util/Annotations.scala

```diff
@@ -135,6 +156,10 @@ case class RegFieldDescMappingAnnotation(
 
 object GenRegDescsAnno {
 
+  var GLOBAL_EXTERNAL_INTERRUPTS = "global-external-interrupts"
+  var LOCAL_EXTERNAL_INTERRUPTS = "local-external-interrupts"
+  var LOCAL_INTERRUPTS_STARTING_NUMBER = 16 /* TODO the ISA specfication reserves the first 12 interrupts but
```

Comment:
1. Why are you adding these constants to the `GenRegDecsAnno` object, when they are not used in RegDesc annotations? 
2. Why are they `var`s if they are constants?
3. I found where the 16 is coming from: https://github.com/freechipsproject/rocket-chip/blob/master/src/main/scala/tile/Interrupts.scala#L59
4. I think you should add a `def localInterruptOffset: Int = 16` function to `HasExternalInterrupts` and use it to fill in the value there and also in your annotation

Reply:
1,2 fixed

3. Andrew said to dot he following:

Add the following to Core.scala
HasCoreParameters

def firstLocalInterrupt = new MIP().getWidth - coreParams.nLocalInterrupts

Add to Interrupts.scala

import freechips.rocketchip.tile._

replace

GenRegDescsAnno.LOCAL_INTERRUPTS_STARTING_NUMBER

with

HasCoreParameters.firstLocalInterrupt

But HasCoreParameters.firstLocalInterrupt is not available in Annotations.scala...